### ISSUE TITLE
BaseUrl option is not goes into amd config, then importScript in monaco worker access the wrong path

### ISSUE THREAD

**Issue by Jack-Works** [superRaytin/react-monaco-editor#61]
![image](https://user-images.githubusercontent.com/5390719/31499375-2c98ab14-af97-11e7-9071-509783f185ba.png)
![image](https://user-images.githubusercontent.com/5390719/31499462-71a5ed98-af97-11e7-8991-44c33ed32603.png)

**Comment by domoritz** [comment]
I have no idea what the problem is. Can you try to describe it a bit more and provide some code to reproduce the issue?

**Comment by Jack-Works** [comment]
I set a baseUrl =`/static` in react-monaco-editor, then by unknown reason, it does not pass into Monaco.
When the web worker of monaco try to load language service file, `this.options.baseUrl` is empty(but I need `/static`), so load fails.

**Comment by Jack-Works** [comment]
I'll try to reproduce it later, not now, sorry

**Comment by jeffmcaffer** [comment]
I appear to be having the same problem using the Diff editor at a non-root location in my app. For example I have the following in the `render` of a Component that shows up in my app at http://localhost:3000/foo/bar.

```html
      <MonacoDiffEditor
        height='400'
        original='current value'
        value='new value'
        requireConfig={{ baseUrl: '/' , url: '/vs/loader.js' }}
      />
```

without the `requireConfig` vs/loader is fetched from http://localhost:3000/foo/vs/loader.js (notice the `foo` in the path). With `requireConfig.url` set as show above, vs/loader.js loads fine. However, then the following in `diff.js` fails for the same reason (loading the module from the wrong place)

```javascript
        // Load monaco
        context.require(['vs/editor/editor.main'], function () {
          _this3.initMonaco();
        });
```
`requireConfig.baseUrl` looked like it would solve the problem but it does not seem to be taking effect during module loading.

Digging around a bit more I noticed the following in `diff.js`

```javascript
          if (requireConfig.paths && requireConfig.paths.vs) {
            context.require.config(requireConfig);
          }
```

Seems like the `requireConfig` prop needs to have a `path.vs` value for it to work. I updated my code to the following and it works:
```html
      <MonacoDiffEditor
        height='400'
        original='currentCuration'
        value='curationValue'
        requireConfig={{ baseUrl: '/', paths: { vs: 'vs' }, url: '/vs/loader.js' }}
      />
```
Ideally one would just use `requireConfig: {{ baseUrl: '/' }}` and that would address the whole loading location issue. Actually, it might even make sense to fix the location of the Monaco code since there is likely only one place to get it for any given app.

**Comment by ThePainnn** [comment]
+1 exact same problem

**Comment by tx4x** [comment]
same here, baseUrl doesn't work

**Comment by tx4x** [comment]
a fix is easy and here : https://github.com/tx4x/react-monaco-editor/blob/master/src/editor.js#L116

**Comment by tx4x** [comment]
@Jack-Works , c'mon; it's copy & paste 4 lines only :-)

**Comment by Jack-Works** [comment]
Alright, I found that this problem is getting fixed in a commit between 0.12.0 and 0.14.1, but I don't know which one. So if anyone found this problem, just update.
